test/MatrixProductOperator_test.jl

```diff
@@ -0,0 +1,86 @@
+@testset "MatrixProductOperator" begin
+    using Tenet: TensorNetwork, Operator, Closed, Open, bounds, MatrixProductOperator
+
+    @testset "Types" begin
+        @test MatrixProductOperator <: Operator
+        @test all(T -> MatrixProductOperator{T} <: Operator, [Open, Closed])
+        @test all(T -> MatrixProductOperator{T} <: Operator{T}, [Open, Closed])
+        @test all(B -> bounds(MatrixProductOperator{B}) == B, [Open, Closed])
+    end
+
+    @testset "Constructor" begin
+        @test begin
+            arrays = [rand(3, 1, 1), rand(3, 1, 1)]
+            MatrixProductOperator(arrays) isa TensorNetwork{MatrixProductOperator{Open}}
+        end
+
+        @testset "`Open` boundary" begin
+            # product operator
+            @test begin
+                arrays = [rand(3, 1, 1), rand(3, 3, 1, 1), rand(3, 1, 1)]
+                MatrixProductOperator{Open}(arrays) isa TensorNetwork{MatrixProductOperator{Open}}
+            end
+
+            @test begin
+                arrays = [rand(1, 3, 1), rand(3, 1, 3, 1), rand(3, 1, 1)]
+                MatrixProductOperator{Open}(arrays, order = (:l, :i, :r, :o)) isa TensorNetwork{MatrixProductOperator{Open}}
+            end
+        end
+
+        @testset "`Closed` boundary" begin
+            # product state
+            @test begin
+                arrays = [rand(1, 1, 2, 2), rand(1, 1, 2, 2), rand(1, 1, 2, 2)]
+                MatrixProductOperator{Closed}(arrays) isa TensorNetwork{MatrixProductOperator{Closed}}
+            end
+            
+            @test begin
+                arrays = [rand(3, 1, 1, 3), rand(3, 1, 1, 3), rand(3, 1, 1, 3) ]
+                MatrixProductOperator{Closed}(arrays, order = (:l, :i, :o, :r)) isa TensorNetwork{MatrixProductOperator{Closed}}
+            end
+        end
+    end
+
+    @testset "Initialization" begin
+        @test begin
+            rand(MatrixProductOperator{Open}, 6, 2, 2, 4) isa TensorNetwork{MatrixProductOperator{Open}}
+        end
+
+        @test begin
+            rand(MatrixProductOperator{Open}, 8, 2, 3, 4) isa TensorNetwork{MatrixProductOperator{Open}}
+        end
+
+        @test begin
+            rand(MatrixProductOperator{Open}, 1, 2, 2, 1) isa TensorNetwork{MatrixProductOperator{Open}}
+        end
+
+        @test begin
+            rand(MatrixProductOperator{Open}, 1, 3, 3, 1) isa TensorNetwork{MatrixProductOperator{Open}}
+        end
+
+        @test begin
+            rand(MatrixProductOperator{Open}, 1, 1, 1, 1) isa TensorNetwork{MatrixProductOperator{Open}}
+        end
+
```

Comment:
These tests are failing. It seems it's because there is only one tensor and the `vinds` labels are poorly defined. I would remove this block of code and maybe raise an exception when `n=1`, since I don't know if we want to support a `MatrixProductOperator` with a single tensor. If we want, we can simply add a condition to not include   virtual labels. 
```suggestion

```

Reply:
I think @mofeing was looking into this last night, but didn't receive feedback

Reply:
But yes, why would we have MPO with one tensor only? Then its simply "a tensor"

Reply:
Exactly, this is what I think as well. Furthermore, this problem also appears in `MatrixProductState`, and I think the best solution is to throw an exception. What do you think, @mofeing? I can do a PR for the `MatrixProductState` case, and include a test.

Reply:
Added "@test_skip" for those tests.